src/test/java/com/dnd/runus/infrastructure/persistence/domain/challenge/ChallengeAchievementRepositoryImplTest.java

```diff
@@ -72,4 +88,49 @@ void saveAchievement() {
         assertNotNull(saved);
         assertTrue(saved.isSuccess());
     }
+
+    @DisplayName("러닝 레코드 리스트로 ChallengeAchievement의 id 찾기")
+    @Test
+    void findIdsByRunningRecordsTest() {
+        // given
+        List<Long> ids = new ArrayList<>();
+        for (int i = 0; i < 2; i++) {
+            ids.add(challengeAchievementRepository
+                    .save(new ChallengeAchievement(challenge, savedRunningRecords.get(i), true))
+                    .ChallengeAchievementId());
+        }
+
+        // then
+        List<Long> resultIds = challengeAchievementRepository.findIdsByRunningRecords(savedRunningRecords);
+
+        // when
+        assertNotNull(resultIds);
+        assertThat(resultIds.size()).isEqualTo(ids.size());
+        assertTrue(resultIds.containsAll(ids));
+    }
+
+    @DisplayName("Challenge achievement 삭제 : id 리스트로")
+    @Test
+    void deleteAchievementByIdsTest() {
+        // given
+        List<Long> ids = new ArrayList<>();
+        for (int i = 0; i < 2; i++) {
+            ids.add(challengeAchievementRepository
+                    .save(new ChallengeAchievement(challenge, savedRunningRecords.get(i), true))
+                    .ChallengeAchievementId());
+        }
+
+        // when
+        challengeAchievementRepository.deleteByIds(ids);
+
+        // then
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+
+        CriteriaQuery<ChallengeAchievementEntity> query = criteriaBuilder.createQuery(ChallengeAchievementEntity.class);
+        List<ChallengeAchievementEntity> selectAll = em.createQuery(
+                        query.select(query.from(ChallengeAchievementEntity.class)))
+                .getResultList();
+
+        assertThat(selectAll.size()).isEqualTo(0);
```

Comment:
그게 중요한 건 아닌데, isEmpty를 사용하면 의미가 좀더 명확해질 것 같아요

```suggestion
        assertTrue(selectAll.isEmpty());
```

Reply:
서비스 단 구현 하면서 같이 수정해둘게요!